course_flow/tests/test_models.py

```diff
@@ -1144,198 +1161,108 @@ def test_add_node_permissions_no_authorship(self):
+            self.assertEqual(created_node.classification, node.classification)
+            #Add the strategy through add-strategy
             response = self.client.post(
-                reverse("course_flow:add-component-to-program"),
-                {"componentPk": component.id, "programPk": program.id},
+                reverse("course_flow:add-strategy"),
+                {"workflowPk": wf.id, "strategyPk": strategy.id},
             )
-            self.assertEqual(response.status_code, 200)
-            new_component = Component.objects.latest("pk")
+            self.assertEqual(Strategy.objects.all().count(), 3)
             self.assertEqual(
-                ComponentProgram.objects.get(
-                    program=program, component=new_component
-                ).rank,
-                0,
+                StrategyWorkflow.objects.filter(workflow=wf).count(), 2
+            )
+            created_strategy = StrategyWorkflow.objects.get(
+                workflow=wf,rank=0
+            ).strategy
+            self.assertEqual(
+                StrategyWorkflow.objects.get(workflow=wf,strategy=created_strategy).rank, 0
+            )
+            self.assertNotEqual(created_strategy, strategy)
+            self.assertEqual(created_strategy.title, strategy.title)
+            self.assertEqual(created_strategy.description, strategy.description)
+            self.assertEqual(created_strategy.author, strategy.author)
+            self.assertEqual(created_strategy.author, user)
+            self.assertEqual(created_strategy.parent_strategy, strategy)
+            self.assertFalse(created_strategy.is_original)
+
+            self.assertEqual(Node.objects.all().count(), 3)
+            new_created_node = NodeStrategy.objects.get(
+                strategy=created_strategy
+            ).node
+            self.assertEqual(
+                NodeStrategy.objects.get(strategy=created_strategy).rank, 0
+            )
+            self.assertNotEqual(new_created_node, node)
+            self.assertEqual(new_created_node.title, node.title)
+            self.assertEqual(new_created_node.description, node.description)
+            self.assertEqual(new_created_node.author, node.author)
+            self.assertEqual(new_created_node.author, user)
+            self.assertEqual(new_created_node.parent_node, created_node)
+            self.assertFalse(new_created_node.is_original)
+            self.assertEqual(
+                new_created_node.work_classification, node.work_classification
+            )
+            self.assertEqual(
+                new_created_node.activity_classification,
+                node.activity_classification,
             )
-            self.assertEqual(new_component, component)
-            self.assertTrue(new_component.content_object.is_original)
+            self.assertEqual(new_created_node.classification, node.classification)
+            Strategy.objects.all().delete()
```

Comment:
Remove these mass delete statements.  Django's test framework does this already.

Reply:
That's actually not (strictly) true. Note that I've combined the activity, course, and program into a single loop here. This ensures the nodes and strategies get purged at the end of each iteration (usually this is done automatically at the end of the function).

Reply:
My bad, I mistakenly read this as being outside of the for loop. Thought they were just being deleted at the end of the function for no reason.